angular/projects/spark-angular/src/lib/components/sprk-masthead/directives/sprk-masthead-nav-collapsible/sprk-masthead-nav-collapsible.directive.ts

```diff
@@ -0,0 +1,52 @@
+import { Directive, Input, HostBinding, ElementRef } from '@angular/core';
+import { uniqueId } from 'lodash';
+@Directive({
+  selector: '[sprkMastheadNavCollapsible]',
+})
+export class SprkMastheadNavCollapsibleDirective {
+  /**
+   * @ignore
+   */
+  constructor(public ref: ElementRef) {}
+
+  /**
+   * The value supplied will be assigned to the `data-id` attribute on the
+   * component. This is intended to be used as a selector for automated
+   * tools. This value should be unique per page.
+   */
+  @HostBinding('attr.data-id')
+  @Input()
+  idString: string;
+
+  /**
+   * The value supplied will be assigned to the
+   * `data-analytics` attribute on the element.
+   * Intended for an outside
+   * library to capture data.
+   */
+  @HostBinding('attr.data-analytics')
+  @Input()
+  analyticsString: string;
+
+  /**
+   * The value supplied will be assigned to the `id` attribute on the
+   * component. This is intended to be used as a hook for the collapsible
+   * nav button.
+   */
+  @HostBinding('attr.id')
+  @Input()
+  id = uniqueId(`sprk_masthead_nav_collapsible_`);
+
+  /**
+   * If `true`, styles will be applied to hide the nav.
+   */
+  @Input() isCollapsed = true;
+
+  @HostBinding('class.sprk-c-Masthead__nav-collapsible')
+  mastheadNavCollapsibleClass = true;
+
+  @HostBinding('class.sprk-c-Masthead__nav-collapsible--is-collapsed')
```

Comment:
Resolved in chat. We are going to not include additionalClasses on directives for now because it could be confusing to users for them when deciding to use additionalClasses or the regular class attribute.